core/src/main/java/com/linecorp/armeria/server/file/HttpFileServiceConfig.java

```diff
@@ -36,25 +38,43 @@
 
     private final HttpVfs vfs;
     private final Clock clock;
-    private final int maxCacheEntries;
+    private final Optional<String> entryCacheSpec;
     private final int maxCacheEntrySizeBytes;
     private final boolean serveCompressedFiles;
     private final boolean autoIndex;
     private final HttpHeaders headers;
 
-    HttpFileServiceConfig(HttpVfs vfs, Clock clock, int maxCacheEntries, int maxCacheEntrySizeBytes,
+    HttpFileServiceConfig(HttpVfs vfs, Clock clock, Optional<String> entryCacheSpec, int maxCacheEntrySizeBytes,
                           boolean serveCompressedFiles, boolean autoIndex, HttpHeaders headers) {
         this.vfs = requireNonNull(vfs, "vfs");
         this.clock = requireNonNull(clock, "clock");
-        this.maxCacheEntries = validateMaxCacheEntries(maxCacheEntries);
+        this.entryCacheSpec = validateEntryCacheSpec(entryCacheSpec);
         this.maxCacheEntrySizeBytes = validateMaxCacheEntrySizeBytes(maxCacheEntrySizeBytes);
         this.serveCompressedFiles = serveCompressedFiles;
         this.autoIndex = autoIndex;
         this.headers = requireNonNull(headers, "headers");
     }
 
-    static int validateMaxCacheEntries(int maxCacheEntries) {
-        return validateNonNegativeParameter(maxCacheEntries, "maxCacheEntries");
+    static Optional<String> validateEntryCacheSpec(Optional<String> entryCacheSpec) {
+        if (!entryCacheSpec.isPresent() || "off".equals(entryCacheSpec.get())) {
+            return Optional.empty();
+        }
+        try {
+            CaffeineSpec.parse(entryCacheSpec.get());
+        } catch (Exception e) {
+            throw new IllegalArgumentException("invalid cache spec: " + entryCacheSpec, e);
+        }
+        return entryCacheSpec;
+    }
+
+    @Deprecated
```

Comment:
This was package-private before, so we don't have to put `@Deprecated` here?

Reply:
Oh, My mistake. I will fix it.